module/modmanager/manager_test.go

```diff
@@ -1430,3 +1431,125 @@ func TestModularDiscovery(t *testing.T) {
 		})
 	}
 }
+
+func TestFirstRun(t *testing.T) {
+	ctx := context.Background()
+	logger, logs := logging.NewObservedTestLogger(t)
+
+	exePath := rtestutils.BuildTempModuleWithFirstRun(t, "module/testmodule")
+	modCfg := config.Module{
+		Name:    "test-module",
+		ExePath: exePath,
+	}
+	parentAddr := setupSocketWithRobot(t)
+	opts := modmanageroptions.Options{
+		UntrustedEnv: false,
+	}
+	mgr := setupModManager(t, ctx, parentAddr, logger, opts)
+
+	// Subtests must be run sequentially in the given order.
+	// Re-ordering the tests is likely to cause failures.
+
+	t.Run("fails", func(t *testing.T) {
+		t.Setenv("VIAM_TEST_FAIL_RUN_FIRST", "1")
+		err := mgr.FirstRun(ctx, modCfg)
+		test.That(t, err, test.ShouldNotBeNil)
+
+		testutils.WaitForAssertion(t, func(tb testing.TB) {
```

Comment:
if we wait for FirstRun to complete, do we actually need the WaitForAssertions in this set of tests?

Reply:
no i don't think we do, will remove to simplify things